lib/win_checker.rb

```diff
@@ -0,0 +1,26 @@
+require_relative 'tic_tac_toe_rules'
+
+class WinChecker
+
+  def initialize(board)
+    @board = board
+    @full_sets = get_full_sets(@board.get_side_length)
+  end
+
+  def has_player_won?(token)
+    player_moves_set = @board.get_move_set_for_player(token)
+
+    @full_sets.any? { |full_set|
+      full_set.subset?(player_moves_set) 
+    }
+  end
+
+  private
+
+  def get_full_sets(side_length)
+    t = TicTacToeRules.new(side_length)
```

Comment:
Every time this method is called, it creates a new TicTacToeRules class.  You should initialize this in the constructor.  Or, better yet, pass it into the constructor.

Reply:
I modified the WinChecker to take in a TicTacToeRules object in the constructor.